notebooks/ir_scattered_light_calwf3_corrections/Correcting_for_Scattered_Light_in_IR_Exposures_Using_calwf3_to_Mask_Bad_Reads.ipynb

```diff
@@ -0,0 +1,726 @@
+    "integ_time = np.zeros(shape=(NSAMP))\n",
+    "for i in range(1, NSAMP+1):\n",
+    "        integ_time[i-1] = ima_orig[('TIME', i)].header['PIXVALUE']\n",
+    "integ_time = integ_time[::-1]\n",
+    "\n",
+    "dt = np.diff(integ_time)\n",
+    "\n",
+    "final_time = integ_time[-1]\n",
+    "\n",
+    "if (len(reads) > 0):\n",
+    "    for read in reads:\n",
+    "        index = len(integ_time)-read-1 #because the reads are stored in reverse order\n",
+    "        final_time -= dt[index]\n",
+    "        \n",
+    "        \n",
+    "print(f'The final exposure time after reprocessing is {final_time}.')\n",
+    "image_new[0].header['EXPTIME'] = final_time\n",
+    "image_new.flush()\n",
+    "image_new.close()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, note that the reprocessed FLT product now includes a larger sky background in science pixels corresponding to IR \"blobs\", regions of reduced sensitivity due particulate matter on the channel selection mechanism (CSM) mirror([section 7.5 of the WFC3 Data Handbook](https://hst-docs.stsci.edu/wfc3dhb/chapter-7-wfc3-ir-sources-of-error/7-5-blobs)). \n",
+    "\n",
+    "The method for correcting WFC3/IR images described in the notebook Correcting for Scattered Light in WFC3/IR Exposures: Manually Subtracting Bad Reads (O'Connor, in prep.) provides FLT products without blobs, but that do include cosmic rays."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 6. Drizzling Nominal and Reprocessed FLT Products <a id=\"drizzle\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In our example we use an exposure ('icqtbbbxq') from image association 'ICQTBB020' acquired in visit BB of program [14037](https://www.stsci.edu/hst/observing/program-information). This visit consists of two orbits of two exposures each, and we now download the three other FLTs in the visit ('icqtbbc0q_flt.fits', 'icqtbbbrq_flt.fits', 'icqtbbbtq_flt.fits') and the pipeline drizzled DRZ product. \n",
+    "\n",
+    "To produce a clean DRZ image (without blob residuals), we can drizzle the four FLTs together (from the nominal exposures and the reprocessed exposure), replacing pixels flagged as blobs with those from the dithered (nominal) image.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_list = Observations.query_criteria(obs_id=OBS_ID)\n",
+    "\n",
+    "Observations.download_products(data_list['obsid'], project='CALWF3',  \n",
+    "                mrp_only=False, productSubGroupDescription=['FLT','DRZ'])\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nominal_file_ids = [\"icqtbbc0q\", \"icqtbbbrq\", \"icqtbbbtq\"]\n",
+    "#nominal_file_ids = [\"icxt27hoq\"]\n",
```

Comment:
Is this supposed to be for the observation set with the earth limb at the end?

Reply:
yes!